File: year2020/month08/day0817/gebal_0817.c
int main() {
	int n, cache[100] = { 0 }, cnt = 0;
	while (1)
	{
		scanf(" %d", &n);
		if (n == 0) {
			printf("%d\n", cnt);
			for (int i = 0; i < cnt; i++) {
				if (cache[i] % 2 == 0) {
					printf("%d ", cache[i] / 2);
				}
				else {
					printf("%d ", cache[i] * 2);
				}
			}
			break;
		}
		cache[cnt] = n;
		cnt++;
	}
}

File: year2020/month08/day0817/gebal_0817.py
cache = []
while True:
    n = int(input())
    if n == 0:
        print(len(cache))
        [print(int(x/2), end=' ') if x % 2 == 0 else print(x*2, end=' ') for x in cache]
        break
    cache.append(n)
